src/types/env.ts

```diff
@@ -4,14 +4,75 @@ import "dotenv/config";
 import { StandardValidator } from "typebox-validators";
 
 /**
- * Define sensitive environment variables here.
- *
- * These are fed into the worker/workflow as `env` and are
- * taken from either `dev.vars` or repository secrets.
- * They are used with `process.env` but are type-safe.
+ * We can restrict which updates the BotFather bot will receive.
  */
-export const envSchema = T.Object({});
+const allowedUpdates = T.Object({
+  message: T.String(),
+  poll: T.String(),
+  edited_message: T.String(),
+  channel_post: T.String(),
+  edited_channel_post: T.String(),
+  business_connection: T.String(),
+  business_message: T.String(),
+  edited_business_message: T.String(),
+  deleted_business_messages: T.String(),
+  message_reaction_count: T.String(),
+});
 
-export const envValidator = new StandardValidator(envSchema);
+export const env = T.Object({
+  /**
+   * The token for the bot given by the BotFather.
+   */
+  BOT_TOKEN: T.String(),
+  /**
+   * The url to forward updates to.
+   */
+  BOT_WEBHOOK: T.String(),
+  /**
+   * The secret to use when forwarding updates.
+   */
+  BOT_WEBHOOK_SECRET: T.String(),
+  /**
+   * Ids of the users who are allowed to use admin commands.
+   */
+  BOT_ADMINS: T.Transform(T.Unknown())
+    .Decode((str) => (Array.isArray(str) ? str.map(Number) : [Number(str)]))
+    .Encode((arr) => arr.toString()),
+  /**
+   * Which updates the bot should receive, defaults to all.
+   */
+  ALLOWED_UPDATES: T.Optional(T.Array(T.KeyOf(allowedUpdates))),
+  /**
+   * The supabase instance url for storing chats, sessions, etc.
+   */
+  SUPABASE_URL: T.String(),
+  /**
+   * The supabase service key for storing chats, sessions, etc.
+   */
+  SUPABASE_SERVICE_KEY: T.String(),
+  /**
+   * Obtained from https://my.telegram.org/apps
+   */
+  TELEGRAM_APP_ID: T.Transform(T.Unknown())
+    .Decode((str) => Number(str))
+    .Encode((num) => num.toString()),
+  /**
+   * Obtained from https://my.telegram.org/apps
+   */
+  TELEGRAM_API_HASH: T.String(),
```

Comment:
What is api hash

Reply:
I'm assuming it's the Telegram equivalent of `APP_PRIVATE_KEY` as it's immutable and account-bound, can't seem to create another after the inital.